middle_end/inlining_transforms.ml

```diff
@@ -272,7 +302,7 @@ let inline_by_copying_function_declaration ~env ~r
        copied.  We add these bindings using [Let] around the new
        set-of-closures declaration. *)
     let free_vars, free_vars_for_lets =
-      fold_over_projections_of_vars_bound_by_closure ~closure_id_being_applied
+      fold_over_projections_of_vars_bound_by_closures ~closure_id_being_applied
```

Comment:
You should restrict this to the 'required_functions', which are computed just after.

Reply:
Done. As a side note, `required_functions` is probably too large:
it takes into account every function that is referenced while we
are interested in functions such as there is a direct call to them
(will open another PR with this change).